src/generate/peripheral.rs

```diff
@@ -104,75 +104,274 @@ pub fn render(
     Ok(out)
 }
 
+#[derive(Clone, Debug)]
 struct RegisterBlockField {
     field: syn::Field,
     description: String,
     offset: u32,
     size: u32,
 }
 
+#[derive(Clone, Debug)]
+struct Region {
+    fields: Vec<RegisterBlockField>,
+    offset: u32,
+    end: u32,
+}
+
+impl Region {
+    /// Find the longest common prefix of the identifier names
+    /// for the fields in this region, if any.
+    fn common_prefix(&self) -> Option<String> {
```

Comment:
To that implementation there may be a need to be add a grouping of numerical and non-alpha characters to make  `foo1` and `foo12` have a common prefix of `foo`, however, this would really depend on the context of the register